integrade/api.py

```diff
@@ -0,0 +1,216 @@
+# coding=utf-8
+"""Client for working with Cloudigrade's API.
+
+This module provides a flexible API client for talking with the cloudigrade
+server, allowing the user to customize how return codes are handled depending
+on the context.
+
+"""
+from json import JSONDecodeError
+from pprint import pformat
+from urllib.parse import urljoin, urlunparse
+
+from integrade import config, exceptions
+
+import requests
+from requests.exceptions import HTTPError
+
+
+def raise_error_for_status(response):
+    """Generate an error message and raise HTTPError for bad return codes.
+
+    :raises: ``requests.exceptions.HTTPError`` if the response status code is
```

Comment:
FWIW, in the rest of the cloudigrade project we've been using the "Google style" for Python docstrings, but I wouldn't oppose if you want to stick with the more traditional "RST style" docstrings here. Eventually we should converge on a common format, and even though the Google format is easier on my human eyes, RST is more well-structured for machine parsing if we want to chew it through Sphinx.

Examples of the Google style:
https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html

Reply:
I would check in with @elyezer about this. We use a tool that consumes our doc strings and determines which ones are test cases for the project and spits this information into Polarian aka "plutonium". I'm not sure what the consequences of mixing styles would be.

Reply:
We can use the google style for the source code, but the test cases we need them to follow the RST format since we have tools that parses the docstring and fill information on a test case management system.

That said, to be consistent following RST format everywhere would be more appropriate.